src/transforms/kubernetes/watch_client.rs

```diff
@@ -0,0 +1,469 @@
+use crate::dns::Resolver;
+use futures::{future::Future, stream::Stream};
+use http::{header, status::StatusCode, uri, Request, Uri};
+use hyper::{client::HttpConnector, Body, Chunk};
+use hyper_openssl::HttpsConnector;
+use k8s_openapi::{
+    api::core::v1::{Pod, WatchPodForAllNamespacesResponse},
+    apimachinery::pkg::apis::meta::v1::WatchEvent,
+    RequestError, Response, ResponseError, WatchOptional,
+};
+use openssl::ssl::{SslConnector, SslConnectorBuilder, SslMethod};
+use snafu::{ResultExt, Snafu};
+
+/// Config which could be loaded from kubeconfig or local kubernetes cluster.
+/// Used to build WatchClient which is in turn used to build Stream of metadata.
+pub struct ClientConfig {
+    resolver: Resolver,
+    token: Option<String>,
+    server: Uri,
+    ssl: Option<SslConnectorBuilder>,
+    node_name: Option<String>,
+}
+
+impl ClientConfig {
+    pub fn build(mut self) -> Result<WatchClient, BuildError> {
+        let mut http = HttpConnector::new_with_resolver(self.resolver.clone());
+        http.enforce_http(false);
+
+        let ssl = if let Some(ssl) = self.ssl.take() {
+            ssl
+        } else {
+            SslConnector::builder(SslMethod::tls()).context(OpenSslError)?
+        };
+
+        let https = HttpsConnector::with_connector(http, ssl).context(OpenSslError)?;
+
+        let client = hyper::Client::builder().build(https);
```

Comment:
I think we want to reuse `sinks::util::http::https_client` here.

Reply:
We can't since `HttpsConnector` in `sinks::util::http::https_client` is from `hyper_tls`, and here we are using `hyper-openssl`.

Reply:
Ok, for now its fine to use our sinks version but we have this one open https://github.com/timberio/vector/issues/1402 to change that one to be openssl based, so if we just base it off of `https_client` now it will use openssl once we fix that issue.